Sources/AppcuesKit/Data/Models/Experience/ExperienceComponent+Sticky.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  ExperienceComponent+Sticky.swift
+//  AppcuesKit
+//
+//  Created by Matt on 2023-02-07.
+//  Copyright © 2023 Appcues. All rights reserved.
+//
+
+import Foundation
+
+// swiftlint:disable large_tuple
+
+extension ExperienceComponent {
+    /// Extract sticky stacks while filtering those from the body component.
+    func divided() -> (body: ExperienceComponent, stickyTop: ExperienceComponent?, stickyBottom: ExperienceComponent?) {
+        let (body, stickyTopComponents, stickyBottomComponents) = self.split()
+
+        return (
+            // `body` is `nil` only if `self` is sticky, which means there's no root content to be displayed so populate something empty.
+            body ?? ExperienceComponent.spacer(SpacerModel(id: UUID(), spacing: nil, style: nil)),
```

Comment:
makes sense. I took a slightly different approach on Android where I split the content using the children of the root component (if it was a type that had children, which it is typically vstack) - effectively meaning there was always a root component that was assumed non-sticky. This is a bit more flexible, but I think probably both approaches work ok in practice.

Reply:
Yeah, given our actual experiences having a root VStack, this doesn't matter. I could rewrite this as a `guard` to make it clearer that we really expect the `body` to be there and the Spacer is a last resort fallback. Clearer semantic than the nil-coalescing.